storage/innobase/ut/ut0crc32.cc

```diff
@@ -201,16 +228,26 @@ ut_crc32_64_low_hw(
 	uint64_t	data)
 {
 	uint64_t	crc_64bit = crc;
-
+#ifdef __WIN__
+#ifdef _WIN64
+	crc_64bit = _mm_crc32_u64(crc_64bit, data);
+#else
+	crc = _mm_crc32_u32(crc, static_cast<uint32_t>(data));
+	crc_64bit = _mm_crc32_u32(crc, static_cast<uint32_t>(data >> 32));
+#endif
```

Comment:
I'd prefer something like:
```
#ifdef _WIN64
...
#elif defined(__WIN__)
...
#else
...
#endif
```
But it is not strong preference. @vaintroub may have different opinion. :)

Reply:
I have a different opinion indeed .
First, __WIN__ is something we do not use in new code, we use more standard _WIN32 :)  __WIN__ is from MySQL code base in the 90ties.  Background - we should not use anything with double undescores, it is there for the compiler writers.

anything else is less important to me. Pedantically,  _WIN32/_WIN64 should probably be _M_IX86 and _M_X64,  but we can save this pedantic stuff until we port of ARM

Reply:
I am thinking I should wrap the code under "_MSVC_VER" as we can build mariadb with mingw-win32. the current implementation doesn't consider that.

Reply:
if you mean _MSC_VER rather than _MSVC_VER,  this is correct. So probably do not compile on mingw, at least we have not tested for long time, but there are inquiries from time to time.